File: AntiClimacus/DataAccessLayer/Entities/DonationRequest.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace DataAccessLayer.Entities
{
    public class DonationRequest
    {
        public int Id { get; set; }
        public int DonorId { get; set; }
        public string RecipientName { get; set; }
        public DateTime Date { get; set; }
        public int StatusId { get; set; }
        public bool Active { get; set; }

        public ICollection<BloodDonation> BloodDonations { get; set; }
        public Status Status { get; set; }
        public Donor Donor { get; set; }
    }
}

File: DonatorSange-Angular/src/app/models/donationRequest.ts
import { UnitStatus } from "./unitStatus";
import { getLocaleDateTimeFormat } from "@angular/common";

export class DonationRequest{
    id:number;
    recipientName:string;
    date:Date;
    active:boolean;
    status:UnitStatus;
    static initialize(){
        let donationRequest = new DonationRequest();
        donationRequest.id = 0;
        donationRequest.recipientName = null;
        donationRequest.date = new Date();
        donationRequest.active = false;
        donationRequest.status = UnitStatus.initialize();
        return donationRequest;
    }
}